TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift

```diff
@@ -91,18 +91,37 @@ class TradeItTradingTicketViewController: TradeItViewController, TradeItSymbolSe
     }
 
     @IBAction func previewOrderTapped(sender: UIButton) {
-        self.ezLoadingActivityManager.show(text: "Previewing Order", disableUI: true)
-
-        order.preview(onSuccess: { previewOrder, placeOrderCallback in
-            self.ezLoadingActivityManager.hide()
-            self.delegate?.tradeItTradingTicketViewController(self,
-                previewOrder: previewOrder,
-                placeOrderCallback: placeOrderCallback)
-            
-        }, onFailure: { errorResult in
-            self.ezLoadingActivityManager.hide()
-            self.alertManager.show(tradeItErrorResult: errorResult, onViewController: self, withLinkedBroker: self.order.linkedBrokerAccount!.linkedBroker, onFinished: {})
-        })
+        self.ezLoadingActivityManager.show(text: "Authenticating", disableUI: true)
+        if let linkedBroker = self.order.linkedBrokerAccount?.linkedBroker {
+            linkedBroker.authenticateIfNeeded(
+                onSuccess: {
+                    self.ezLoadingActivityManager.updateText(text: "Previewing Order")
+                    self.order.preview(onSuccess: { previewOrder, placeOrderCallback in
+                        self.ezLoadingActivityManager.hide()
+                        self.delegate?.tradeItTradingTicketViewController(self,
+                            previewOrder: previewOrder,
+                            placeOrderCallback: placeOrderCallback)
+                        
+                        }, onFailure: { errorResult in
+                            self.ezLoadingActivityManager.hide()
+                            self.alertManager.show(tradeItErrorResult: errorResult, onViewController: self, withLinkedBroker: linkedBroker, onFinished: {})
+                    })
+                },
+                onSecurityQuestion: { (securityQuestion: TradeItSecurityQuestionResult, answerSecurityQuestion: (String) -> Void, cancelSecurityQuestion: () -> Void) -> Void in
+                    self.ezLoadingActivityManager.hide()
+                    self.alertManager.show(
+                        securityQuestion: securityQuestion,
+                        onViewController: self,
+                        onAnswerSecurityQuestion: answerSecurityQuestion,
+                        onCancelSecurityQuestion: cancelSecurityQuestion
+                    )
+                },
+                onFailure: { (tradeItErrorResult: TradeItErrorResult) -> Void in
```

Comment:
We don't need to prefix the variables with `tradeIt` everywhere or give the type signatures/return value for the function